### PR TITLE
test: add playwright for e2e tests

### LINKED ISSUE #688: Tests
Add tests for modV's core application.

### PR DESCRIPTION
**Pull request by 2xAA**
fix: #688

Started this again - fixed the window launch order which was preventing reliable testing previously.
Seems pretty solid this time around.

Not tested on any other OS than macOS.
To run tests:

```bash
yarn
yarn electron:build
yarn test:e2e
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,27 +71,33 @@
     "@babel/core": "^7.0.0-0",
     "@babel/plugin-proposal-nullish-coalescing-operator": "^7.16.0",
     "@babel/plugin-proposal-optional-chaining": "^7.16.0",
+    "@playwright/test": "^1.31.2",
     "@semantic-release/git": "^9.0.0",
     "@vue/cli-plugin-babel": "^4.5.15",
     "@vue/cli-plugin-eslint": "^3.12.1",
     "@vue/cli-service": "^3.12.1",
     "@vue/eslint-config-prettier": "^4.0.1",
+    "asar": "^3.2.0",
```
Comment (2xAA) [inline package.json]: May not be needed

File: package.json [hunk 2]
```diff
@@ -71,27 +71,33 @@
...
+    "playwright": "^1.31.2",
+    "playwright-core": "^1.31.2",
     "sass-loader": "^7.3.1",
     "text-loader": "0.0.1",
     "vue-cli-plugin-electron-builder": "^2.0.0",
     "vue-template-compiler": "^2.6.10",
     "vuex-localstorage": "^1.0.0",
-    "worker-loader": "^2.0.0"
+    "worker-loader": "^2.0.0",
+    "xvfb-maybe": "^0.2.1"
```
Comment (2xAA) [inline package.json]: May not be needed

File: tests/e2e/main.spec.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
+import { expect, test } from "@playwright/test";
+// import {
+//   clickMenuItemById,
+//   ipcMainCallFirstListener,
+//   ipcRendererCallFirstListener,
+//   ipcMainInvokeHandler,
+//   ipcRendererInvoke
+// } from "electron-playwright-helpers";
```
Comment (2xAA) [inline tests/e2e/main.spec.js:7]: Can remove, was following a tutorial

File: tests/e2e/main.spec.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  expect(await activeModules.count()).toBe(1);
+
+  const activeModule = await activeModules
+    .first()
+    .locator(".active-module__title");
+  const activeModuleTitle = await activeModule.textContent();
+  expect(activeModuleTitle.trim()).toBe("Ball");
+});
+
+// test("trigger IPC listener via main process", async () => {
```
Comment (2xAA) [inline tests/e2e/main.spec.js:11]: Can remove, was following a tutorial

File: tests/e2e/pageObjectModel/index.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        });
+        // capture console messages
+        page.on("console", msg => {
+          console.log(msg.text());
+        });
+      });
+    });
+
+    test.afterAll(() => {
+      // Playwright, why are you so bad at timings?
```
Comment (2xAA) [inline tests/e2e/pageObjectModel/index.js]: Unhappy about this hack, but it seems needed.

Luckily it seems like it's only teardown timing, so doesn't affect the running of the tests until the last.

File: tests/e2e/pageObjectModel/index.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    if (!isReady) {
+      setTimeout(() => {
+        this.waitUntilModVReady(resolver);
+      }, 500);
+
+      return;
+    }
+
+    resolver();
+    resolver = undefined;
```
Comment (2xAA) [inline tests/e2e/pageObjectModel/index.js]: Probably not needed

**Comment by 2xAA** [comment]
Screenshot 2023-05-22 at 20 53 20

**Review by TimPietrusky — CHANGES_REQUESTED** [review]
Everything is awesome, just one problem with the `groups/groupNameCanBeChanged.spec.js`. As we found out during the call, its related to my MacOS version.

**Comment by 2xAA** [comment]
@TimPietrusky groupNameCanBeChanged is now fixed!

**Comment by 2xAA** [comment]
Screenshot 2023-05-25 at 08 12 57

**Pull request merged**